uikit/views/src/main/java/com/sygic/maps/uikit/views/navigation/lanes/SimpleLanesView.kt

```diff
@@ -0,0 +1,176 @@
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+
+package com.sygic.maps.uikit.views.navigation.lanes
+
+import android.content.Context
+import android.graphics.Color
+import android.graphics.PorterDuff
+import android.graphics.drawable.LayerDrawable
+import android.util.AttributeSet
+import android.view.Gravity
+import android.view.View
+import android.view.ViewGroup
+import android.widget.ImageView
+import android.widget.LinearLayout
+import androidx.annotation.ColorInt
+import androidx.core.content.ContextCompat
+import androidx.core.util.Pools
+import com.sygic.maps.uikit.views.R
+import com.sygic.maps.uikit.views.common.extensions.dpToPixels
+import com.sygic.maps.uikit.views.common.extensions.getColorFromAttr
+import com.sygic.maps.uikit.views.navigation.lanes.data.SimpleLanesData
+
+private const val POOL_SIZE = 10
+
+/**
+ * [SimpleLanesView] displays actual lanes on the road in front of the vehicle as arrows
+ * and highlights those which follows the direction of the current route.
+ */
+class SimpleLanesView @JvmOverloads constructor(
+    context: Context,
+    attrs: AttributeSet? = null,
+    defStyleAttr: Int = R.attr.lanesViewStyle,
+    defStyleRes: Int = R.style.SygicLanesViewStyle
+) : LinearLayout(context, attrs, defStyleAttr, defStyleRes) {
+
+    /**
+     * Provides lane data for the [SimpleLanesView] to show. Single [item][SimpleLanesData]
+     * in the array represents single line.
+     */
+    var lanes: Array<SimpleLanesData> = emptyArray()
+        set(value) {
+            field = value
+            setLanesInternal(value)
+        }
+
+    private var layoutMargin: Int = 0
+    private var layoutMarginTop: Int = 0
+    private var layoutMarginBottom: Int = 0
+    private var layoutMarginStart: Int = 0
+    private var layoutMarginEnd: Int = 0
+
+    @ColorInt
+    private val highlightedColor = context.getColorFromAttr(R.attr.navigationTextColorTertiary)
+    private val laneViewSize = resources.dpToPixels(24f).toInt()
```

Comment:
Toto 24f a ten 8f padding by mohli být reálné resourcy, které by se načítali, alespoň by se dali v případě potřeby overridnout následně v app ;)

Reply:
noo nepodporujeme to ani pri signposte - tiez su tam marginy natvrdo 8dp

Reply:
size - 24 som dal do dimens

Reply:
what? :) v signpostu mám někde natrvdo v kódu 8f? :)

Reply:
nie v kode, ale v xml ano

Reply:
to ano :)